sdk/storage/azure-storage-file-datalake/src/test/java/com/azure/storage/file/datalake/LargeFileTest.groovy

```diff
@@ -8,143 +8,101 @@ import com.azure.core.http.HttpResponse
 import com.azure.core.http.policy.HttpPipelinePolicy
 import com.azure.storage.common.ParallelTransferOptions
 import com.azure.storage.common.implementation.Constants
-import com.azure.storage.common.policy.StorageSharedKeyCredentialPolicy
+import com.azure.storage.common.test.shared.TestEnvironment
+import com.azure.storage.common.test.shared.TestHttpClientType
 import com.azure.storage.common.test.shared.extensions.LiveOnly
 import com.azure.storage.file.datalake.models.DataLakeStorageException
+import org.apache.commons.lang3.SystemUtils
 import reactor.core.publisher.Flux
 import reactor.core.publisher.Mono
-import spock.lang.Ignore
-import spock.lang.Requires
+import spock.lang.IgnoreIf
+import spock.lang.ResourceLock
 import spock.lang.Unroll
 
 import java.nio.ByteBuffer
 import java.util.concurrent.atomic.AtomicLong
-import java.util.function.BiFunction
 
+@LiveOnly
+@ResourceLock("LargeFileTest")
+@IgnoreIf({ TestEnvironment.getInstance().httpClientType == TestHttpClientType.OK_HTTP || SystemUtils.IS_OS_MAC }) // https://github.com/Azure/azure-sdk-for-java/issues/23221
```

Comment:
- mac agents are not collocated with storage account, so disabling there
- okhttp doesn't like payloads larger than int.max...